### ISSUE TITLE
Animation looks differently with different rendering mode

### ISSUE THREAD

**Issue by Banck** [airbnb/lottie-ios#1876]
Hello! 
I'm trying **Lottie 4.0.1** and animation broken with rendering mode automatic. I attached json file and screenshot how this file looks:
Web Preview:
<img width="412" alt="image" src="https://user-images.githubusercontent.com/6259254/208851927-49388887-65ab-4316-a25f-f51b05db3337.png">

iOS with mainThread:
<img width="360" alt="image" src="https://user-images.githubusercontent.com/6259254/208851882-fd66284a-dc1c-4154-8ac7-6c8ac1ab5406.png">

iOS with automatic rendering mode:
<img width="376" alt="image" src="https://user-images.githubusercontent.com/6259254/208851826-5999d26c-3cab-4e9e-a43c-8e0ceba6dfcd.png">

As you can see on the Web Preview animation looks cool, but on iOS with mainThread there is some glitches on the top, on iOS with automatic rendering mode animation is broken.

**Comment by calda** [comment]
Thanks for the report. I've investigated some and understand why this issue is happening, but don't know a good way to fix it. Leaving some notes for future reference:

The `Tower Poles` shape has a structure like this:
 - Group
   - Larger Rectangle
   - Shape transform
 - Group
   - Smaller Rectangle
   - Shape transform
  - Red Fill

In the main thread rendering engine, the two rectangles are combined into a single `CGPath` that is that rendered with the red fill. Drawing the two shapes as a single `CGPath` causes the small rectangle to be subtracted from the larger rectangle, which is effective rendered as a red outlined triangle.

The Core Animation rendering engine converts each Group into a `CAShapeLayer` drawing a single `CGPath` with the red fill. This means both paths are opaque and are just drawn on top of each other (rather than smaller rectangle being subtracted from the larger rectangle). Each path has its own transform, and we currently only apply transforms to `CAShapeLayer`s, so we don't currently have a way to correctly apply the transform before then merging the shapes.

I'm not sure there's a way to fix this without substantially rewriting how groups are handled by the Core Animation rendering engine, and I'm not sure we have a good way to detect this unsupported case without having too many false-positives.

**Comment by Banck** [comment]
> Thanks for the report. I've investigated some and understand why this issue is happening, but don't know a good way to fix it. Leaving some notes for future reference:
> 
> The `Tower Poles` shape has a structure like this:
> 
> * Group
>   
>   * Larger Rectangle
>   * Shape transform
> * Group
>   
>   * Smaller Rectangle
>   * Shape transform
> * Red Fill
> 
> In the main thread rendering engine, the two rectangles are combined into a single `CGPath` that is that rendered with the red fill. Drawing the two shapes as a single `CGPath` causes the small rectangle to be subtracted from the larger rectangle, which is effective rendered as a red outlined triangle.
> 
> The Core Animation rendering engine converts each Group into a `CAShapeLayer` drawing a single `CGPath` with the red fill. This means both paths are opaque and are just drawn on top of each other (rather than smaller rectangle being subtracted from the larger rectangle). Each path has its own transform, and we currently only apply transforms to `CAShapeLayer`s, so we don't currently have a way to correctly apply the transform before then merging the shapes.
> 
> I'm not sure there's a way to fix this without substantially rewriting how groups are handled by the Core Animation rendering engine, and I'm not sure we have a good way to detect this unsupported case without having too many false-positives.
> 
> A workaround would be to either:
> 
> 1. manually use the main thread rendering engine for this animation
> 2. update the animation to just use a single path shape for the `Tower Poles` item (specifically drawing just the tower outline) instead of multiple shapes that have to be combined at runtime.

Hey! 
Thanks for your comment. So if Core animation engine can’t handle animations with such structure - .automatic mode should select main thread, isn’t it?

**Comment by calda** [comment]
Yes, that's the intended behavior of `.automatic`. We'd just have to manually implement a step that checks for this specific unsupported configuration.

**Comment by calda** [comment]
Here's another example of this issue: https://github.com/airbnb/lottie-ios/issues/2085

**Comment by calda** [comment]
https://github.com/airbnb/lottie-ios/issues/2613#issuecomment-3679083551
